examples/Seabug/Contract/MarketPlaceListNft.purs

```diff
@@ -35,14 +35,17 @@ marketPlaceListNft = do
       plutusData <- liftedM "marketPlaceListNft: Cannot get datum from hash"
         (getDatumByHash datumHash)
       case fromData plutusData of
-        Just (NftData nftData) -> do
+        Just (MarketplaceDatum { getMarketplaceDatum: curr' /\ name' }) -> do
           policy' <- liftedM "marketPlaceListNft: Cannot apply arguments"
             (policy nftData.nftCollection mp)
           curr <- liftedM "marketPlaceListNft: Cannot get CurrencySymbol"
             (scriptCurrencySymbol policy')
-          name <- liftedM "marketPlaceListNft: Cannot hash old token"
+          name <- liftedM "marketPlaceListNft: Cannot hash token"
             (mkTokenName nftData.nftId)
-          pure $ valueOf amount curr name == one
+          pure
+            $ (curr == curr')
+            && (name == name')
+            && (valueOf amount curr name == one)
```

Comment:
I added in parenthesis, otherwise it looks a little silly with the formatter unless we go over 80 characters.

Reply:
^ Ignore.